### PR TITLE
refactor(iot-device): Move product info to client options

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -85,6 +84,11 @@ internal class InternalClient : IDisposable
                 throw new InvalidOperationException("Plug and Play is not supported over the HTTP transport.");
             }
 
+            var productInfo = new ProductInfo
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:87]: Why is this created here?

Comment (drwill-ms) [reply iothub/device/src/InternalClient.cs:87]: Ah never mind.

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/tests/IotHubDeviceClientTests.cs [hunk 2]
```diff
@@ -1032,15 +1032,6 @@ public void DeviceClientOnConnectionClosedInvokeHandlerAndRecoveryForStatusChang
             Assert.AreEqual(ConnectionStatusChangeReason.No_Network, statusChangeReason);
         }
 
-        [TestMethod]
-        public void ProductInfoStoresProductInfoOk()
-        {
-            const string userAgent = "name/version (runtime; os; arch)";
-            IotHubDeviceClient client = IotHubDeviceClient.CreateFromConnectionString(FakeConnectionString);
-            client.ProductInfo = userAgent;
-            Assert.AreEqual(userAgent, client.ProductInfo);
```
Comment (abhipsaMisra) [inline iothub/device/tests/IotHubDeviceClientTests.cs:1041]: this test wasn't really testing anything, so I've removed it

File: iothub/device/src/IotHubModuleClient.cs [hunk 3]
```diff
@@ -476,21 +467,23 @@ private async Task<MethodResponse> InvokeMethodAsync(Uri uri, MethodRequest meth
                     };
                 }
 
-                var context = new PipelineContext()
+                // TODO (abmisr) - fix when ClientOptions is made available in PipelineContext (InternalClient)
```
Comment (abhipsaMisra) [inline iothub/device/src/IotHubModuleClient.cs:470]: This is a TODO for the next PR

**Pull request merged**